mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -335,6 +335,6 @@ def _to_configspace_configs(self, configurations: pd.DataFrame) -> List[ConfigSp
             List of ConfigSpace configurations.
         """
         return [
-            ConfigSpace.Configuration(self.optimizer_parameter_space, values=config.to_dict())
-            for (_, config) in configurations.iterrows()
+            ConfigSpace.Configuration(self.optimizer_parameter_space, values=config._asdict())
+            for config in configurations.itertuples(index=False)
```

Comment:
That's neat. We should use it in other similar situations. I did not know that `namedtuple._asdict()` is not private.

P.S. could you please check that pylint is not complaining about accessing a private method - and if it does, add a corresponding `# pylint disable:...` comment?

Reply:
I was also surprised it is not a private method.

pylint looks good on my end